kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigSecret.java

```diff
@@ -174,8 +195,10 @@ private static List<NamedFilterDefinition> filterDefinitions(Clusters cluster, C
             var filterCr = filterResourceFromRef(cluster, context, filterCrRef);
             if (filterCr.getAdditionalProperties().get("spec") instanceof Map<?, ?> spec) {
                 String type = (String) spec.get("type");
-                Object config = spec.get("config");
-                return new NamedFilterDefinition(filterDefinitionName, type, config);
+                SecureConfigInterpolator.InterpolationResult interpolationResult = interpolateConfig(spec);
+                context.managedDependentResourceContext().put(SECURE_VOLUME_KEY, interpolationResult.volumes());
+                context.managedDependentResourceContext().put(SECURE_VOLUME_MOUNT_KEY, interpolationResult.mounts());
```

Comment:
There's a bit more work required here for aggregating volumes and mounts over multiple filters.